apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -257,6 +296,80 @@ export class ProjectBuildCache {
     return success;
   }
 
+  private async _tryCollectPathsToCacheAsync(terminal: Terminal): Promise<IPathsToCache | undefined> {
+    const projectFolderPath: string = this._project.projectFolder;
+    const outputFolderNamesThatExist: boolean[] = await Promise.all(
+      this._projectOutputFolderNames.map((outputFolderName) =>
+        FileSystem.existsAsync(path.join(projectFolderPath, outputFolderName))
+      )
+    );
+    const filteredOutputFolderNames: string[] = [];
+    for (let i: number = 0; i < outputFolderNamesThatExist.length; i++) {
+      if (outputFolderNamesThatExist[i]) {
+        filteredOutputFolderNames.push(this._projectOutputFolderNames[i]);
+      }
+    }
+
+    let encounteredEnumerationIssue: boolean = false;
+    function symbolicLinkPathCallback(entryPath: string): void {
+      terminal.writeError(`Unable to include "${entryPath}" in build cache. It is a symbolic link.`);
+      encounteredEnumerationIssue = true;
+    }
+
+    const outputFilePaths: string[] = [];
+    for (const filteredOutputFolderName of filteredOutputFolderNames) {
+      if (encounteredEnumerationIssue) {
+        return undefined;
+      }
+
+      const outputFilePathsForFolder: AsyncIterableIterator<string> = this._getPathsInFolder(
+        terminal,
+        symbolicLinkPathCallback,
+        filteredOutputFolderName,
+        projectFolderPath + path.sep + filteredOutputFolderName
+      );
+
+      for await (const outputFilePath of outputFilePathsForFolder) {
+        outputFilePaths.push(outputFilePath);
+      }
+    }
+
+    if (encounteredEnumerationIssue) {
+      return undefined;
+    }
+
+    return {
+      filteredOutputFolderNames,
+      outputFilePaths
+    };
+  }
+
+  private async *_getPathsInFolder(
+    terminal: Terminal,
+    symbolicLinkPathCallback: (path: string) => void,
+    posixPrefix: string,
+    folderPath: string
+  ): AsyncIterableIterator<string> {
+    const folderEntries: fs.Dirent[] = await LegacyAdapters.convertCallbackToPromise(fs.readdir, folderPath, {
+      withFileTypes: true
+    });
+    for (const folderEntry of folderEntries) {
+      const entryPath: string = `${posixPrefix}/${folderEntry.name}`;
+      if (folderEntry.isSymbolicLink()) {
+        symbolicLinkPathCallback(entryPath);
+      } else if (folderEntry.isDirectory()) {
+        yield* this._getPathsInFolder(
+          terminal,
+          symbolicLinkPathCallback,
+          entryPath,
+          folderPath + path.sep + folderEntry.name
```

Comment:
I can't think of a good reason to not always use `/` here.

Reply:
Yup - seems to work fine.